src/GitHub.VisualStudio/TeamExplorer/Connect/GitHubConnectSection.cs

```diff
@@ -156,11 +256,59 @@ public void Login()
             StartFlow(UIControllerFlow.Authentication);
         }
 
+        public void OpenRepository()
+        {
+            var old = Repositories.FirstOrDefault(x => String.Equals(Holder.ActiveRepo?.RepositoryPath, x.LocalPath, StringComparison.CurrentCultureIgnoreCase));
+            // open the solution selection dialog when the user wants to switch to a different repo
+            // since there's no other way of changing the source control context in VS
+            if (SelectedRepository != old)
```

Comment:
It's on purpose, the equality operators are overriden in SimpleRepositoryModel to perform comparisons based on content (because the instances are different, maybe not their content). There's comments in SimpleRepositoryModel to that effect.